libtiledbsoma/src/reindexer/reindexer.cc

```diff
@@ -46,11 +46,17 @@ KHASH_MAP_INIT_INT64(m64, int64_t)
 
 namespace tiledbsoma {
 
-void IntIndexer::map_locations(const int64_t* keys, int size, int threads) {
+void IntIndexer::map_locations(
+    const int64_t* keys, size_t size, size_t threads) {
     map_size_ = size;
+
+    // Handling edge cases
     if (size == 0) {
         return;
     }
+    if (threads == 0) {
+        throw std::runtime_error("Re-indexer thread count cannot be zero.");
+    }
 
     LOG_DEBUG(fmt::format(
```

Comment:
any reason to have duplicate logging here - same information is in both statements....

Reply:
fixed